qdax/core/neuroevolution/mdp_utils.py

```diff
@@ -179,6 +175,48 @@ def scoring_function(
     )
 
 
+@partial(
+    jax.jit,
+    static_argnames=(
+        "episode_length",
+        "play_reset_fn",
+        "play_step_fn",
+        "behavior_descriptor_extractor",
+    ),
+)
+def stochastic_scoring_function(
+    policies_params: Genotype,
+    random_key: RNGKey,
+    episode_length: int,
+    play_reset_fn: Callable[[RNGKey], brax.envs.State],
+    play_step_fn: Callable[
+        [brax.envs.State, Params, RNGKey, brax.envs.Env],
+        Tuple[brax.envs.State, Params, RNGKey, QDTransition],
+    ],
+    behavior_descriptor_extractor: Callable[[QDTransition, jnp.ndarray], Descriptor],
+) -> Tuple[Fitness, Descriptor, Dict[str, Union[jnp.ndarray, QDTransition]], RNGKey]:
+    """Evaluates policies contained in policies_params in parallel.
+    The play_reset_fn function allows to define purely stochastic environments.
+    To use the reset function from the environment set it to "partial(env.reset)".
```

Comment:
Why `partial(env.reset)`? Wouldn't `env.reset` work?

Reply:
Yes `env.reset` would work indeed.
Also, is it useful to add to this comment that the user can also use it for the deterministic case by setting `play_reset_fn` such that it always returns the same `init_state`?

Reply:
I changed the function name to "reset_based_scoring_function" following a discussion with Bryan, and updated its description.